### PR TITLE
Multilevel ibfe 4

### PR DESCRIPTION
**Pull request by drwells**
### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

**Comment by drwells** [comment]
This PR lets us refine patches based on what level elements should be on. Here is a picture, taken from one of the new tests:

visit0002

**Comment by drwells** [comment]
The tests depend on #1118 - we need to merge that first.

File: ibtk/src/lagrangian/FEDataManager.cpp [hunk 1]
```diff
@@ -2587,6 +2599,9 @@ FEDataManager::applyGradientDetector(const Pointer<BasePatchHierarchy<NDIM> > hi
             for (unsigned int e_idx = 0; e_idx < num_active_patch_elems; ++e_idx)
             {
                 Elem* const elem = patch_elems[e_idx];
+                // If the element should already be on the current level then we
+                // do not need to mark any cells for refinement
+                if (getPatchLevel(elem) <= level_number) continue;
```
Comment (drwells) [inline ibtk/src/lagrangian/FEDataManager.cpp:2604]: This is the important change in the PR.

**Review by boyceg — APPROVED** [review]
Looks good!

**Pull request merged**